opm/simulators/wells/WellGroupHelpers.cpp

```diff
@@ -963,7 +964,13 @@ computeNetworkPressures(const Network::ExtNetwork& network,
 
                     if (well.isInjector() || !well_state.isOpen(wellname)) continue;
                     const Scalar efficiency = well.getEfficiencyFactor(/*network*/ true) * well_state.getGlobalEfficiencyScalingFactor(wellname);
-                    node_inflows[node][BlackoilPhases::Vapour] += well_state.getALQ(wellname) * efficiency;
+                    Scalar alq = 0.0;
+                    const auto& well_index = well_state.index(well.name());
+                    if (well_index.has_value() && well_state.wellIsOwned(well_index.value(), well.name())) {
+                        alq += well_state.well(wellname).alq_state.get();
+                    }
+                    alq = comm.sum(alq);
```

Comment:
I am not totally familiar how the parallel working here and did not test it, but from the code, it looks worrying. 

Very likely, you should `comm.sum()` outside the for loop `for (const std::string& wellname : group.wells())` to avoid the extra communications. 

And then update the `node_inflows[node][BlackoilPhases::Vapour]` outside the for loop with the summation. 

Please let me know what you think.

Reply:
Thanks! I moved the communication outside the well loop.